src/test/java/org/example/controllers/ReaderControllerTest.java

```diff
@@ -0,0 +1,174 @@
+import static org.example.util.Util.getResponseForInvalidFieldsNewReaderDTO;
+import static org.example.util.Util.getTestReaders;
+import static org.example.util.Util.setIdForTestReaders;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+
+@AutoConfigureMockMvc
+@ExtendWith(SpringExtension.class)
+@WebMvcTest(ReaderController.class)
+@Import(DateTimeFormatterConfiguration.class)
+class ReaderControllerTest {
+
+  private static final String REQUEST_PATH = "/api/v1/readers";
+
+  @Autowired
+  MockMvc mvc;
+  @Autowired
+  ObjectMapper objectMapper;
+  @MockBean
+  ReaderService readerService;
+  @Autowired
+  DateTimeFormatter dateTimeFormatter;
+
+  @Test
+  void shouldReturnNotFoundWhenReaderNotPresent() throws Exception {
+    var idTen = 10L;
+    when(readerService.findById(idTen)).thenReturn(Optional.empty());
+    mvc.perform(get(REQUEST_PATH + "/" + idTen)).andExpect(status().isNotFound());
+  }
+
+  @Test
+  void shouldReturnReaderWithId1() throws Exception {
+    var idOne = 1L;
+    when(readerService.findById(idOne)).thenReturn(Optional.of(getFistReader()));
+    var mvcResult = mvc.perform(get(REQUEST_PATH + "/" + idOne)).andExpect(status().isOk())
+        .andReturn();
+
+    var contentAsString = mvcResult.getResponse().getContentAsString();
+    var readerFromResponse = objectMapper.readValue(contentAsString, Reader.class);
+
+    assertAll(() -> assertEquals(idOne, readerFromResponse.getId()),
+        () -> assertEquals("reader", readerFromResponse.getName()));
+
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = {"Patric", "John Random", "Mike Douglas"})
+  void shouldSaveNewReader(String name) throws Exception {
+    var readerDTO = new NewReaderDTO(name);
+    var reader = new Reader(name);
+    reader.setId(5L);
+    when(readerService.addNewReader(any(Reader.class))).thenReturn(reader);
+
+    MvcResult mvcResult = mvc.perform(post(REQUEST_PATH).contentType("application/json")
+            .content(objectMapper.writeValueAsString(readerDTO))).andExpect(status().is(201))
+        .andReturn();
+
+    var content = mvcResult.getResponse().getContentAsString();
+    var readerFromResponse = objectMapper.readValue(content, Reader.class);
+
+    assertAll(() -> assertEquals(readerFromResponse.getId(), reader.getId()),
+        () -> assertEquals(name, readerFromResponse.getName()));
+
+  }
```

Comment:
Please verify that the 201 response has the correct "Location" for the created resource